osf_tests/test_queued_mail.py

```diff
@@ -183,3 +185,15 @@ def test_remind_prereg_presend_deleted_draft(self, mock_mail, user, prereg):
         prereg.deleted = timezone.now()
         prereg.save()
         assert not mail.send_mail()
+
+    def test_disabled_queued_emails_not_sent_if_switch_active(self, user):
+        SwitchFactory(name=DISABLE_ENGAGEMENT_EMAILS, active=True)
+        assert self.queue_mail(mail=NO_ADDON, user=user) is False
+        assert self.queue_mail(mail=NO_LOGIN, user=user) is False
+        assert self.queue_mail(mail=WELCOME_OSF4M, user=user) is False
+        assert self.queue_mail(mail=NEW_PUBLIC_PROJECT, user=user) is False
+
+    def test_disabled_triggered_emails_not_sent_if_switch_active(self):
```

Comment:
This isn't really the appropriate place for this test, but it seemed like the best option given there isn't a test_mails file.